app/main.py

```diff
@@ -1,6 +1,22 @@
-from typing import Callable
+from functools import wraps
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> Any:
+        func_key = func.__name__
+        key = (func_key, args, frozenset(kwargs.items()))
+
+        if key in cache_dict:
+            print("Getting from cache")
+            return cache_dict[key]
+
+        result = func(*args, **kwargs)
+        cache_dict[key] = result
+        print("Calculating new result")
+        return result
+
+    return wrapper
```

Comment:
There is a `py-cache-decorator.py` file that is empty in your pull request. You should remove it from the pull request since it should contain only the `app/main.py` file.